pkg/telemetryquerylanguage/tql/README.md

```diff
@@ -112,16 +112,37 @@ Booleans can be either:
 
 Operators determine how the two Values are compared.  The valid operators are:
 
-- Equal (`==`). Equal (`==`) checks if the left and right Values are equal, using Go's `==` operator.
-- Not Equal (`!=`).  Not Equal (`!=`) checks if the left and right Values are not equal, using Go's `!=` operator.
+- Equal (`==`). Tests if the left and right Values are equal (see the Comparison Rules below).
+- Not Equal (`!=`).  Tests if the left and right Values are not equal.
+- Less Than (`<`). Tests if left is less than right. For binary values, `false` is considered to be less than `true`.
+- Less Than (`<`). Tests if left is less than right.
+
+
+### Comparison Rules
+
+The table below describes what happens when two Values are compared. Value types are provided by the user of TQL. In this table, `int` refers to all integer types (int, int32, int64), and `float` similarly refers to both `float32` and `float64`. If numeric values are of different types, they are compared as the most general instance required (so `int32` and `int64` are compared as `int64`, while `int64` and `float64` are compared as `float64`), In addition, pointers to `bool`, `int` types, `float` types, and `string` are compared by first dereferencing the pointer.
```

Comment:
Not sure how I feel about making the assumption to dereference pointers during a comparison.  I think we might be making too many assumptions there.  Whats the argument for performing a dereference before comparison?

Reply:
I was trying to follow existing usage. In a lot of cases (perhaps even usually?), the object returned by a getter is a pointer to a value. See the tqltest file, for example. Existing test cases, anyway, pretty much assume pointers. In essence, I added the evaluation of non-pointers to the semantics.

Reply:
If you look through the 3 standard contexts I don't think we have any situations where we return pointers.  The tqltest functions are primarily used to setup test input Path's or expected parsed structures, but the Getter interface implementations normally return regular values.  I think it probably best we don't make any assumptions and compare as close to what the user has configured as we can.